src/cascade/dismod/db/metadata.py

```diff
@@ -0,0 +1,519 @@
+    time_id = Column(None, ForeignKey("time.time_id"))
+    value_prior_id = Column(Integer(), nullable=True)
+    """A prior_id. If null, const_value must not be null."""
+
+    dage_prior_id = Column(Integer(), nullable=True)
+    """A prior_id. If null, const_value must not be null."""
+
+    dtime_prior_id = Column(Integer(), nullable=True)
+    """A prior_id. If null, const_value must not be null."""
+
+    const_value = Column(Float(), nullable=True)
+    """If non-null, specifies the value of the function at the corresponding
+    age and time. It sets lower, upper, and mean to this value and density
+    to uniform. If null, value_prior_id must not be null."""
+
+
+class NSList(Base):
+    __tablename__ = "nslist"
+
+    nslist_id = Column(Integer(), primary_key=True)
+    nslist_name = Column(String(), unique=True)
+
+
+class NSListPair(Base):
+    """Associates a node with a smoothing"""
+
+    __tablename__ = "nslist_pair"
+
+    nslist_pair_id = Column(Integer(), primary_key=True)
+    nslist_id = Column(None, ForeignKey("nslist.nslist_id"))
+    node_id = Column(None, ForeignKey("node.node_id"))
+    smooth_id = Column(None, ForeignKey("smooth.smooth_id"))
+
+
+class RateName(enum.Enum):
+    pini = 0
+    iota = 1
+    rho = 2
+    chi = 3
+    omega = 4
+
+
+class Rate(Base):
+    __tablename__ = "rate"
+
+    rate_id = Column(Integer(), primary_key=True)
+    rate_name = Column(Enum(RateName))
+    parent_smooth_id = Column(None, ForeignKey("smooth.smooth_id"), nullable=True)
+    """If null, then parent rate is always zero and no model variables are
+    allocated for it"""
+
+    child_smooth_id = Column(None, ForeignKey("smooth.smooth_id"), nullable=True)
+    child_nslist_id = Column(None, ForeignKey("nslist.nslist_id"), nullable=True)
+    """If not null, identifies a list of node and smoothing pairs. The
+    node_id for each of the children must appear in the list. The corresponding
+    smoothing is used for that child and the rate corresponding to this
+    row of the rate table."""
+
+
+class MulCovEnum(enum.Enum):
+    rate_value = 0
+    meas_value = 1
+    meas_std = 2
+
+
+class MulCov(Base):
+    """Covariate multipliers and their priors.
+
+    If this is empty, there are no covariate multipliers in the model.
+    """
+
+    __tablename__ = "mulcov"
+
+    mulcov_id = Column(Integer(), primary_key=True)
+    mulcov_type = Column(Enum(MulCovEnum), nullable=False)
+    rate_id = Column(None, ForeignKey("rate.rate_id"), nullable=True)
+    """Determines the rate that this covariate and multiplier affects.
+    If mulcov_type is of type meas_value or meas_std, this must be null."""
+    integrand_id = Column(None, ForeignKey("integrand.integrand_id"), nullable=True)
+    covariate_id = Column(None, ForeignKey("covariate.covariate_id"), nullable=False)
+    smooth_id = Column(None, ForeignKey("smooth.smooth_id"), nullable=True)
+    """If this is null, the covariabe multiplier is always zero and no
```

Comment:
covariabe -> covariate

Reply:
I'm going to blame that one on Drew. Usually typos are my fault so I have to lean into it when they aren't.